src/main/scala/cognite/spark/v1/SdkV1Relation.scala

```diff
@@ -88,20 +88,11 @@ abstract class SdkV1Relation[A <: Product, I](config: RelationConfig, shortName:
       val rowOfAllFields = toRow(item)
       Row.fromSeq(indicesOfRequiredFields.map(idx => rowOfAllFields.get(idx)))
     }
+}
 
-  def insert(rows: Seq[Row]): IO[Unit] =
-    throw new IllegalArgumentException(
-      s"""$shortName does not support the "onconflict" option "abort".""")
-
-  def upsert(rows: Seq[Row]): IO[Unit] =
-    throw new IllegalArgumentException(
-      s"""$shortName does not support the "onconflict" option "upsert".""")
-
-  def update(rows: Seq[Row]): IO[Unit] =
-    throw new IllegalArgumentException(
-      s"""$shortName does not support the "onconflict" option "update".""")
-
-  def delete(rows: Seq[Row]): IO[Unit] =
-    throw new IllegalArgumentException(
-      s"""$shortName does not support the "onconflict" option "delete".""")
+trait WritableRelation {
```

Comment:
Why are we moving this to a trait? The default methods seemed reasonable, what will happen instead now that some of them aren't implemented at all?

Reply:
If they are not implemented for a specific resource type, then those resources will throw their own exception. Moved it into a trait just to make default source cleaner and since each resource will ultimately implement the 4 operations

Reply:
But this default implementation takes care of throwing an appropriate exception? Isn't this cleaner than each resource implementing its own "throw new IllegalArgumentException" for operations it doesn't support?

Reply:
We decided to keep it in a trait bc datapoints and stringdatapoints are the only two that don't implement all 4 operations